src/graph.rs

```diff
@@ -2167,6 +2167,19 @@ pub(crate) fn parse_js_module_from_module_info(
           })
       })
     });
+    let types_res = module_info.jsx_import_source_types.or_else(|| {
```

Comment:
Actually I think this behaviour is wrong. We should only use the `default_jsx_import_source_types` for the `default_jsx_import_source`, but never for a `@jsxImportSource` pragma. However `@jsxImportSourceTypes` should be allowed even if there is no `@jsxImportSource` pragma and we default `default_jsx_import_source`.

@dsherret WDYT?

Reply:
Yeah, I think so, but would you be able to expand on the reasoning why it should behave that way? (Maybe as a comment in the source code)

Reply:
Added reasoning and updated this code to match my description.